services/QuillLMS/app/services/analytics/cache_segment_school_data.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class CacheSegmentSchoolData
+  CACHE_KEYS = [
+    TOTAL_TEACHERS_AT_SCHOOL = 'TOTAL_TEACHERS_AT_SCHOOL',
+    TOTAL_STUDENTS_AT_SCHOOL = 'TOTAL_STUDENTS_AT_SCHOOL',
+    TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL = 'TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL',
+    ACTIVE_TEACHERS_AT_SCHOOL_THIS_YEAR = 'ACTIVE_TEACHERS_AT_SCHOOL_THIS_YEAR',
+    ACTIVE_STUDENTS_AT_SCHOOL_THIS_YEAR = 'ACTIVE_STUDENTS_AT_SCHOOL_THIS_YEAR',
+    TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL_THIS_YEAR = 'TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL_THIS_YEAR'
+  ]
+
+  CACHE_LIFE = 60*60*25
+
+  def initialize(school)
+    @school = school
+  end
+
+  def read(namespace)
+    Rails.cache.read(@school.id, namespace: namespace)
+  end
+
+  def write(data, namespace)
```

Comment:
I used separate `read` and `write` methods instead of `fetch` because I don't want the cache to be written to on a read call if there's nothing present - this data is just for analytics, and since the queries are potentially somewhat expensive I think we'd rather stick with the model where they just get updated by the cron job and not on demand. That use case would only come up when new school admins are created for a school that doesn't already have any, but it still seemed worth avoiding to me. LMK if you feel differently.

I can update the data into one cache key - agreed that it doesn't seem like we'll need to pull it out separately.

Reply:
Got it; makes sense